integration-tests/modules/__tests__/inventory/index.spec.ts

```diff
@@ -608,145 +609,148 @@ medusaIntegrationTestRunner({
-            sku: "MY_SKU2",
-            manage_inventory: true,
-            options: [
-              {
-                option_id: "test-product-new-option",
-                value: "Blue",
-              },
-            ],
-            prices: [{ currency_code: "usd", amount: 100 }],
-          },
-          { headers: { "x-medusa-access-token": "test_token" } }
-        )
+          const [arr, reservationCountPostDelete] =
+            await inventoryService.listAndCountReservationItems({
+              location_id: locationId,
+            })
 
-        const secondVariantId = response.data.product.variants.find(
-          (v) => v.sku === "MY_SKU2"
-        ).id
+          expect(reservationCountPostDelete).toEqual(0)
 
-        const inventoryService = appContainer.resolve("inventoryService")
-        const variantInventoryService = appContainer.resolve(
-          "productVariantInventoryService"
-        )
+          const [, inventoryLevelCountPostDelete] =
+            await inventoryService.listAndCountInventoryLevels({
+              location_id: locationId,
+            })
 
-        const invItem2 = await inventoryService.createInventoryItem({
-          sku: "123456",
+          expect(inventoryLevelCountPostDelete).toEqual(0)
         })
 
-        await variantInventoryService.attachInventoryItem(
-          variantId,
-          invItem2.id,
-          2
-        )
-        await variantInventoryService.attachInventoryItem(
-          secondVariantId,
-          invItem2.id,
-          2
-        )
+        it("should remove the product variant associations when deleting an inventory item", async () => {
+          const secondVariantId = "test-2"
+          const variantId = "test"
 
-        expect(
-          await variantInventoryService.listInventoryItemsByVariant(variantId)
-        ).toHaveLength(2)
+          const remoteLinks = appContainer.resolve(
+            ContainerRegistrationKeys.REMOTE_LINK
+          )
+          const remoteQuery = appContainer.resolve(
+            ContainerRegistrationKeys.REMOTE_QUERY
+          )
+
+          await remoteLinks.create([
+            {
+              productService: {
+                variant_id: variantId,
+              },
+              inventoryService: {
+                inventory_item_id: invItem.id,
+              },
+            },
+            {
+              productService: {
+                variant_id: secondVariantId,
+              },
+              inventoryService: {
+                inventory_item_id: invItem.id,
+              },
+            },
+          ])
 
-        expect(
-          await variantInventoryService.listInventoryItemsByVariant(
-            secondVariantId
+          let links = await remoteQuery(
+            remoteQueryObjectFromString({
+              entryPoint: "product_variant_inventory_item",
+              variables: {
+                filter: { variant_id: [variantId, secondVariantId] },
+                take: 2,
```

Comment:
todo: rm take, if there is more than 2 then it will never be wrong, only if there is less